modules/v2x/v2x_proxy/app/v2x_proxy.cc

```diff
@@ -13,147 +13,211 @@
+  _hdmap = std::make_shared<::apollo::hdmap::HDMap>();
+  _hdmap->LoadMapFromFile(FLAGS_hdmap_file_name);
   auto v2x_carstatus_timer_period = ceil(
       (1.0 / static_cast<int>(FLAGS_v2x_carstatus_timer_frequency)) * 1000);
+  ::apollo::cyber::TimerOption v2x_carstatus_timer_option;
+  v2x_carstatus_timer_option.period =
+      static_cast<uint32_t>(v2x_carstatus_timer_period);
+  v2x_carstatus_timer_option.callback = [this]() {
+    this->on_v2x_car_status_timer();
+  };
+  v2x_carstatus_timer_option.oneshot = false;
   v2x_carstatus_timer_.reset(
-      new cyber::Timer(static_cast<uint32_t>(v2x_carstatus_timer_period),
-                       [this]() { this->OnV2xCarStatusTimer(); }, false));
-
+      new ::apollo::cyber::Timer(v2x_carstatus_timer_option));
   os_interface_.reset(new OsInterFace());
   obu_interface_.reset(new ObuInterFaceGrpcImpl());
-
-  x2v_trafficlight_ = std::make_shared<IntersectionTrafficLightData>();
-  v2x_carstatus_ = std::make_shared<CarStatus>();
-
-  if (!x2v_trafficlight_timer_ || !v2x_carstatus_timer_ || !os_interface_ ||
-      !obu_interface_ || !x2v_trafficlight_ || !v2x_carstatus_) {
-    AFATAL << "Create timer or interface failed.";
-    exit(1);
-  }
-
-  hdmap_.reset(new apollo::hdmap::HDMap());
-  std::string map_name = FLAGS_map_dir + "/" + FLAGS_base_map_filename;
-  if (FLAGS_debug_flag) {
-    map_name = FLAGS_hdmap_file_name;
-  }
-  if (hdmap_->LoadMapFromFile(map_name) != 0) {
-    AFATAL << "Failed to load hadmap file: " << FLAGS_hdmap_file_name;
-    return;
-  }
-  AINFO << "load hdmap file: " << FLAGS_hdmap_file_name;
-
-  if (!os_interface_->InitFlag() || !obu_interface_->InitFlag()) {
-    AFATAL << "Failed to init os interface or obu interface";
-    return;
-  }
-  first_flag_reader_ = node_->CreateReader<StatusResponse>(
-      "/apollo/v2x/inner/sync_flag",
-      [this](const std::shared_ptr<const StatusResponse>& msg) {
-        x2v_trafficlight_timer_->Start();
-      });
-  if (first_flag_reader_ == nullptr) {
-    AERROR << "Create sync flag reader failed";
-    exit(1);
-  }
-  // x2v_trafficlight_timer_->Start();
+  recv_thread_.reset(new std::thread([this]() {
+    while (!exit_) {
+      this->recv_trafficlight();
+    }
+  }));
+  CHECK(!!_hdmap && !!v2x_carstatus_timer_ && !!os_interface_ &&
+        !!obu_interface_ && os_interface_->InitFlag() &&
+        obu_interface_->InitFlag());
+  // FIXME: NEED REWRITE PROGRAM TO FIX IT!
+  planning_thread_.reset(new std::thread([this]() {
+    while (!exit_) {
+      this->recv_os_planning();
+    }
+  }));
   v2x_carstatus_timer_->Start();
-
+  //    ::grpc::ServerBuilder builder;
+  //    builder.AddListeningPort(
+  //        FLAGS_grpc_server_host + ":" + FLAGS_grpc_debug_server_port,
+  //        ::grpc::InsecureServerCredentials());
+  //    builder.RegisterService(_m_service_debug.get());
+  //    _m_server = builder.BuildAndStart();
+  /* Monitor timer to obu status */
```

Comment:
Remove unnecessary code or mark TODO if you need to revisit it. Use "//" for sentences within a line and /* ... */ for paragraphs.